Makes/ViewController.m

```diff
@@ -0,0 +1,146 @@
+        
+    } failure:^(NSURLSessionTask *operation, NSError *error) {
+        NSLog(@"Error: %@", error);
+    }];
+
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+    self.makes = [[NSMutableArray alloc]init];
+    self.imagesCache = [[NSCache alloc]init];
+    self.title = @"MAKES";
+    self.view.backgroundColor = [UIColor lightGrayColor];
+    
+    [self loadMakes];
+    
+    UICollectionViewFlowLayout *layout=[[UICollectionViewFlowLayout alloc] init];
+    self.makesCollectionView =[[UICollectionView alloc] initWithFrame:self.view.frame collectionViewLayout:layout];
+    [self.makesCollectionView setDataSource:self];
+    [self.makesCollectionView setDelegate:self];
+    
+    [self.makesCollectionView setBackgroundColor:[UIColor lightGrayColor]];
+    [self.makesCollectionView registerClass:[MakesCollectionViewCell class] forCellWithReuseIdentifier:@"cellIdentifier"];
+    [self.view addSubview:self.makesCollectionView];
+    
+    
+    UIEdgeInsets padding = UIEdgeInsetsMake(10 , 10, 10, 10);
+    
+    [self.makesCollectionView mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.edges.equalTo(self.view).with.insets(padding);
+    }];
+    
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
+{
+    return self.makes.count;
+}
+
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+    
+    MakesCollectionViewCell *cell=(MakesCollectionViewCell*)[collectionView dequeueReusableCellWithReuseIdentifier:@"cellIdentifier" forIndexPath:indexPath];
+    
+    if (!cell) {
+        cell = [[MakesCollectionViewCell alloc] init];
+    }
+    
+    cell.makeImageView.image = nil;
+    
+    Make* make = [self.makes objectAtIndex:indexPath.item];
+    NSString *url = make.imageSmallURL;
+ 
+    [NSURLConnection sendAsynchronousRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:url]]
+                                       queue:[NSOperationQueue mainQueue]
+                           completionHandler:^(NSURLResponse *response, NSData *data, NSError *error) {
+                               cell.makeImageView.image = [UIImage imageWithData:data];
+                           }];
+    cell.backgroundColor = [UIColor whiteColor];
+    cell.makeName.text = make.nameAr;
+    
+    return cell;
+}
+
+- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath
+{
+    MakeDetailViewController* makeDetailVC = [[MakeDetailViewController alloc]init];
+    Make* currentMake = [self.makes objectAtIndex:indexPath.item];
+    makeDetailVC.makeNameEnglishString = currentMake.nameEn;
+    makeDetailVC.makeNameArabicString = currentMake.nameAr;
+    makeDetailVC.makeImageViewLargeURLString = currentMake.imageLargeURL;
```

Comment:
You can send make object for make details view controller instead of sending each 
object individually